specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,223 @@
+<!-- TEMPLATE
+  The purpose of this spec is to describe a new WebView2 feature and its APIs.
+
+  There are two audiences for the spec. The first are people
+  that want to evaluate and give feedback on the API, as part of
+  the submission process. When it's complete
+  it will be incorporated into the public documentation at
+  docs.microsoft.com (https://docs.microsoft.com/en-us/microsoft-edge/webview2/).
+  Hopefully we'll be able to copy it mostly verbatim.
+  So the second audience is everyone that reads there to learn how
+  and why to use this API. 
+-->
+
+
+# Background
+This API will allow users to drop things such as images, text, and links into the WebView as part of a drag/drop operation.
+The reason that we need a separate API for this with composition hosting is because we don't have an HWND to call RegisterDragDrop on.
+The hosting app needs to call RegisterDragDrop on the HWND that contains the WebView and implement IDropTarget so it can forward the
+calls (IDropTarget::DragEnter, DragMove, DragLeave, and Drop) to the WebView.
+
+Additionally, the hosting app also needs to call into IDropTargetHelper before forwarding those calls to us as documented here:
+https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-idroptargethelper
+
+
+# Description
+NotifyDropTargetAction is meant to provide a way for composition hosted WebViews to receive drop events as part of a drag/drop operation.
+It is the hosting application's responsibility to call RegisterDragDrop (https://docs.microsoft.com/en-us/windows/win32/api/ole2/nf-ole2-registerdragdrop)
+on the HWND that contains any composition hosted WebViews and to implement IDropTarget(https://docs.microsoft.com/en-us/windows/win32/api/oleidl/nn-oleidl-idroptarget)
+to receive the corresponding drop events from Ole32:
+
+- IDropTarget::DragEnter
+- IDropTarget::DragMove
+- IDropTarget::DragLeave
+- IDropTarget::Drop
+
+NotifyDropTargetAction consolidates these four functions into one and the COREWEBVIEW2_DROP_TARGET_ACTION enum
+specifies which corresponding IDropTarget function was called.
+
+
+# Examples
+```c++
+// Initialized elsewhere
```

Comment:
I thought we usually don't show how we get the webview controller in our code samples because that's kind of redundant for every single API. Would it make more sense to just not show the first two lines at all here?

Reply:
Ok, I can see how it can be a little confusing. I added m_ to indicate the webViewCompositionController could be a member variable and removed the two lines.